src/lib/LoginFile.js

```diff
@@ -65,9 +65,12 @@ class LoginFile {
     async toObjectUrl() {
         await this._drawPromise;
 
-        return new Promise(resolve => {
+        return new Promise((resolve, reject) => {
             this.$canvas.toBlob(blob => {
-                if (!blob) throw new Error('Cannot generate URL');
+                if (!blob) {
+                    reject(new Error('Cannot generate URL'));
+                    return;
+                }
```

Comment:
TIL throwing in the `new Promise()` callback doesn't reject the promise. The more you know...